### PR TITLE
fix: update BaseFee JSON-RPC

### PR DESCRIPTION
**Pull request by fedekunze**
## Description

This PR removes the event filtering logic on the current BaseFee JSON-RPC endpoint.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/feemarket/keeper/eip1559.go [hunk 1]
```diff
@@ -16,19 +16,24 @@ import (
...
+	// defined in the parameters (DefaultBaseFee if it hasn't been changed by
+	// governance).
 	if ctx.BlockHeight() == params.EnableHeight {
 		return params.BaseFee.BigInt()
 	}
 
 	// get the block gas used and the base fee values for the parent block.
+	// NOTE: this is not the parent's base fee but the current block's base fee,
+	// as it is retrieved from the transient store, which is committed to the
+	// persistent KVStore after EndBlock (ABCI Commit).
```
Comment (fedekunze) [inline x/feemarket/keeper/eip1559.go:36]: cc @jolube

Comment (jolube) [reply x/feemarket/keeper/eip1559.go:36]: Should the variable `parentBaseFee` be renamed in this case?

**Comment by fedekunze** [comment]
@jolube added a short description. I'll add more details as soon as we add more tests to check the behavior

**Review by crypto-facs — APPROVED** [review]
LGTM

**Pull request merged**